stdlib/weak.ml

```diff
@@ -200,8 +200,10 @@ module Make (H : Hashtbl.HashedType) : (S with type data = H.t) = struct
         t.table.(t.rover) <- emptybucket;
         t.hashes.(t.rover) <- [| |];
       end else begin
-        Obj.truncate (Obj.repr bucket) (prev_len + additional_values);
-        Obj.truncate (Obj.repr hbucket) prev_len;
+        let newbucket = weak_create prev_len in
+        blit bucket 0 newbucket 0 prev_len;
```

Comment:
Note: we know that there are only `live` items in the bucket (which is smaller than `prev_len`), but they may be scattered anywhere below `prev_len`, so filling only upto `live` would be invalid (if an item was lying at a position `live < pos < prev_len` in the previous bucket).